x-pack/platform/plugins/shared/lens/public/app_plugin/lens_top_nav.tsx

```diff
@@ -686,16 +678,30 @@ export const LensTopNavMenu = ({
                         />
                       </EuiCallOut>
                     ),
+                    delegatedShareUrlHandler: async () => {
+                      const { shareableUrl, savedObjectURL } = await getShareURL(
```

Comment:
Seems like we generate a short URL even if we're not going to use it? 
Should we return early `if(currentDoc?.savedObjectId) return savedObjectURL.href` ?

Reply:
Thanks for the call out! You're right there's no point in creating the short url where we have no use for it.